tips/TIP-0040/tip-0040.md

```diff
@@ -0,0 +1,733 @@
+              </details>
+              <details>
+                <summary>NFT Address</summary>
+                <blockquote>An Address derived from an NFT ID which can be unlocked by unlocking the corresponding NFT. Defined in <a href='../TIP-0038/tip-0038.md#nft-address'>TIP-38 (NFT Address)</a>.</blockquote>
+              </details>
+              <details>
+                <summary>Multi Address</summary>
+                <blockquote>Defines a Multi Address that consists of addresses with weights and a threshold value. The Multi Address can be unlocked if the cumulative weight of all unlocked addresses is equal to or exceeds the threshold. Defined in <a href='../TIP-0052/tip-0052.md#multi-address'>TIP-52 (Multi Address)</a>.</blockquote>
+              </details>
+              <details>
+                <summary>Restricted Address</summary>
+                <blockquote>An address that contains another address and allows for configuring its capabilities. Defined in <a href='../TIP-0050/tip-0050.md#restricted-address'>TIP-50 (Restricted Address)</a>.</blockquote>
+              </details>
+            </td>
+          </tr>
+        </table>
+      </details>
+    </td>
+  </tr>
+</table>
+
+### Additional Transaction Syntactic Validation Rules
+
+#### Output Syntactic Validation
+
+- `Amount` field must fulfill the [storage deposit requirements](../TIP-0047/tip-0047.md) and must not be `0`.
+- `Amount` field must be ≤ `Token Supply`.
+- It must hold true that `1` ≤ `Unlock Conditions Count` ≤ `1`.
+- `Unlock Condition Type` of an <i>Unlock Condition</i> must define one of the following types:
+  - <i>Address Unlock Condition</i>
+- <i>Unlock Conditions</i> must be sorted in ascending order based on their `Unlock Condition Type`.
+- Syntactic validation of all present unlock conditions must pass.
+- <i>Address Unlock Condition</i> must be present.
+- `Validator ID` must not be zeroed out.
+
+### Additional Transaction Semantic Validation Rules
+
+- Explicit `Delegation ID`: `Delegation ID` is taken as the value of the `Delegation ID` field in the Delegation output.
+- Implicit `Delegation ID`: When a Delegation output is consumed as an input in a transaction and `Delegation ID` field
+  is zeroed out, take the BLAKE2b-256 hash of the `Output ID` of the input as `Delegation ID`.
+- For every non-zero explicit `Delegation ID` on the output side there must be a corresponding Delegation Output on the
+  input side. The corresponding Delegation output has the explicit or implicit `Delegation ID` equal to that of the
+  Delegation output on the output side.
+- Let `Past Bounded Slot Index` be given by `Commitment Index + Max Committable Age` where `Commitment Index` is the
+  slot index of the commitment input.
+- Let `Past Bounded Epoch Index` be the epoch index corresponding to the slot index `Past Bounded Slot Index`.
+- Let `Future Bounded Slot Index` be given by `Commitment Index + Min Committable Age` where `Commitment Index` is the
```

Comment:
Min Committable Age gets never defined

Reply:
Exactly. The problem is that we cannot say everywhere where we use protocol parameters that "`This Other Parameter` is defined in Protocol Parameteres", otherwise this would be a ton of extra text across all TIPs. 
That's why we have a catch-all definition in the building blocks:
> Protocol parameters used throughout this TIP are defined in [TIP-49](../TIP-0049/tip-0049.md).

which is kind of saying "If an explicit parameter definition is missing somewhere in this TIP, go check if it's a protocol parameter."

Does that make sense?